src/lib/visualization/lqp_visualizer.cpp

```diff
@@ -113,6 +113,12 @@ void LQPVisualizer::_build_dataflow(const std::shared_ptr<AbstractLQPNode>& from
   }
 
   std::ostringstream label_stream;
+
+  // use a copy of the stream's default locale with thousand separator
+  const auto& separate_thousands_locale =
+      std::locale(label_stream.getloc(), std::make_unique<SeparateThousands>().release());
```

Comment:
This is UB. The object held by the unique_ptr will be destroyed before you access it. Simply make it a stack variable and take its address if the STL insists on getting a pointer.

Reply:
This leads to a seg fault:
```c++
SeparateThousands separate_thousands;
const auto& separate_thousands_locale = std::locale(label_stream.getloc(), &separate_thousands);
label_stream.imbue(separate_thousands_locale);
```

This works:
```c++
const auto& separate_thousands_locale = std::locale(label_stream.getloc(), new SeparateThousands);
label_stream.imbue(separate_thousands_locale);
```

I haven't found a better solution yet.

Some resources that also use the `new` solution:
https://en.cppreference.com/w/cpp/locale/numpunct
https://stackoverflow.com/a/7277333
https://stackoverflow.com/a/17530535

Reply:
Wow. std::locale::facet does internal reference counting. I am... slightly disgusted.

In that case, `new` seems like the <s>best</s> least awful approach. Please make sure to document that the STL forces this atrocity on us.

Reply:
https://www.reddit.com/r/cpp/comments/4aaorj/questions_regarding_stdcodecvt/